include/nbl/asset/IAssetManager.h

```diff
@@ -286,30 +294,37 @@ class IAssetManager : public core::IReferenceCounted, public core::QuitSignallin
                     reloadParams.restoreLevels = 0u; // make sure it wont turn into infinite recursion
                     reloadParams.reload = true; // TODO (consider): alternative to this flag: another method in override just to let user choose asset for restore
                 }
-                IAssetLoader::SAssetLoadContext ctx(params, file);
-                auto asset = _override->chooseDefaultAsset(bundle, ctx);
-                if (!asset->isAnyDependencyDummy(restoreLevels - 1u))
-                    return bundle;
 
-                auto reloadBundle = getAssetInHierarchy(_file, _supposedFilename, reloadParams, _hierarchyLevel, _override);
+                auto reloadBundle = getAssetInHierarchy_impl<RestoreWholeBundle>(_file, _supposedFilename, reloadParams, _hierarchyLevel, _override);
 
-                bool restoreSucceeded = _override->handleRestore(std::move(asset), bundle, reloadBundle, restoreLevels);
-                if (!restoreSucceeded) // hm? return empty bundle if restore was requested, but did not succeeded? or just return the bundle? (TODO consider)
-                    return {};
+                if constexpr (RestoreWholeBundle)
+                {
+                    IAssetLoader::SAssetLoadContext ctx(params, file);
+                    auto asset = _override->chooseDefaultAsset(bundle, ctx);
+
+                    // user responsible for checking if assets he wanted to be restored are in fact restored
+                    _override->handleRestore(std::move(asset), bundle, reloadBundle, restoreLevels);
+                }
+                else
+                {
+                    // user responsible for checking if assets he wanted to be restored are in fact restored
```

Comment:
if you shift the responsibility for checking if the restore was successful onto the user, we'll just end up with asstons of duplicate code in the loaders for checking that

Reply:
also this is why it would be useful to have a
```cpp
core::smart_refctd_ptr<IAsset> IAssetLoader::IAssetLoaderOverride::handleRestore(core::smart_refctd_ptr<IAsset>&& _chosenAsset, SAssetBundle& _bundle, SAssetBundle& _reloadedBundle, uint32_t _restoreLevels)
```
so that you may do
```cpp
if (asset->isAnyDependencyDummy(restoreLevels))
{
   auto newChosenAsset = _override->handleRestore(std::move(asset), bundle, reloadBundle, restoreLevels);
   if (newChosenAsset->isAnyDependencyDummy(restoreLevels))
      return {};
}
```